src/crxviewer.js

```diff
@@ -887,6 +895,73 @@ var viewFileInfo = (function() {
     return viewFileInfo;
 })();
 
+function parseExtensionFromManifest(manifest) {
+    var id;
+    if (manifest.browser_specific_settings &&
+        manifest.browser_specific_settings.gecko &&
+        (id = manifest.browser_specific_settings.gecko.id)) {
+        return id;
+    }
+
+    if (manifest.applications && manifest.applications.gecko &&
+        (id = manifest.applications.gecko.id)) {
+        return id;
+    }
+
```

Comment:
In case it's a Chrome extension:

```
if (typeof manifest.key === 'string') {
  return publicKeyToExtensionId(manifest.key);
}
```

and a fall back to the pubkey based on the CRX header. A quick 'n dirty way to do so is to assign it to a global variable at https://github.com/Rob--W/crxviewer/blob/35d3d8af52b2775ac5296a42537e10f2b5711945/src/crxviewer.js#L2019 (don't forget to clear it in the `else` case).

So it would be more like:

```
// note: crx_public_key is the global set as described above.
var crxKey = manifest.key || crx_public_key;
if (typeof crxKey == "string") {
  return publicKeyToExtensionId(crxKey);
}
```

Reply:
This is not related to the X.509 or DER formats in Firefox, so I'll leave this to a future patch.